### PR TITLE
add idleTimeout for function

### PR DESCRIPTION
**Pull request by markfisher**

### CODE REVIEW

File: pkg/apis/extensions.sk8s.io/v1/function.go [hunk 1]
```diff
@@ -40,6 +40,8 @@ type FunctionSpec struct {
 	// +optional
 	Output string `json:"output,omitempty"`
 
+	IdleTimeout int `json:"idleTimeout,omitempty"`
```
Comment (ericbottard) [inline pkg/apis/extensions.sk8s.io/v1/function.go:43]: Needs to be pointer to int to be optional.
Everything exposed at k8s level ought to use int32(64) to guarantee portability.

Changing on merge

Comment (ericbottard) [reply pkg/apis/extensions.sk8s.io/v1/function.go:43]: Also, the unit should appear in the name, as described in https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#naming-conventions and https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#units
Changing to IdleTimeoutMs, let's reopen if this doesn't suit you

Comment (markfisher) [reply pkg/apis/extensions.sk8s.io/v1/function.go:43]: Sounds good. I was considering whether it should be in *seconds*. This is probably fine for now, but let me know if you have an opinion.

**Comment by markfisher** [comment]
updated and merged as https://github.com/ericbottard/sk8s-types/commit/20d5cbea7db5d7db04ce7689f588faf37bad0a8c

**Pull request closed without merging**